### PR TITLE
Make a Project Plan table automatically from issues

### PR DESCRIPTION
**Pull request by steven-murray**
This PR adds a script to create a concise project plan table automatically from issues and PRs. It can be run semi-regularly to create the Plan table for talks etc., and our own perusal. However, it will be more useful generally to view the open tasks through the Projects view.

 Addresses #34

### CODE REVIEW

File: make_project_table.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+            # Get step number
+            step_number = step_number_pattern.search(step.title).group()
+            major_step = ".".join(step_number.split(".")[:2])  # only up to X.X
+            description = step.title.split(step_number)[-1].split(":")[-1].split('--')[-1]
+            if description:
+                description = ": " + description
+
+            # Get title
+            title = f"[{major_step}]({step.url})"
```
Comment (mkolopanis) [inline make_project_table.py]: This particular line will only execute in python >=3.4. That won't be a _huge_ issue if you only expect users to have up to date python but if so it is worth stating.

File: make_project_table.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+            # Get simulators used
+            labels = step.labels
+            sims = ", ".join([lbl.name.split(":")[-1] for lbl in labels if lbl.name.split(":")[0] == 'simulator'])
+
+            simcmp = ", ".join([lbl.name.split(":")[-1] for lbl in labels if lbl.name.split(":")[0] == 'simcmp'])
+            anlcmp = ", ".join([lbl.name.split(":")[-1] for lbl in labels if lbl.name.split(":")[0] == 'pipeline'])
+            assigned = ", ".join([f"[@{assgn.login}]({assgn.url})" for assgn in step.assignees])
```
Comment (mkolopanis) [inline make_project_table.py]: same as previous comment about f-string.

File: make_project_table.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                description=description,
+                sims = sims,
+                simcmp = simcmp,
+                anlcmp = anlcmp,
+                assigned = assigned,
+                status_num = status_num
+            )
+
+        tables += f"### [{proj.name}]({proj.url})\n"
+        tables += f"{proj.body}\n\n"
```
Comment (mkolopanis) [inline make_project_table.py]: same comment again :D

File: make_project_table.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        tables += table_header
+        for major_step in sorted(step_dict.keys()):
+            d = step_dict[major_step]
+            tables += f"{d['status']}  |  {d['title']}{d['description']}  | {d['sims']}  | {d['simcmp']} | {d['anlcmp']} | {d['assigned']}  |\n"
```
Comment (mkolopanis) [inline make_project_table.py]: that make 5 instances of this, then it should be updated to work with other pythons or explicit that you must use python >=3.4

File: make_project_table.py (hunk 1 above)
Comment (steven-murray) [reply make_project_table.py]: I'll just make it explicit in the README/script that you need py34+. I think everyone is using that now... and this script is not vital, just useful.

**Comment by steven-murray** [comment]
I would also like to be able to auto-import the file... but I don't think you can. Anyway, it exists in the top-level directory.

**Comment by jaguirre** [comment]
OK, I clearly did not have my environment set up properly.  I think `pygithub` may not be standard, so we should specify you need it.  Also, I'd not previously set up `.personal-github-token`, so I mya have done that wrong.  

I get an error on the line
`repo = g.get_repo("HERA-Team/hera-validation")`
so I don't know how generic that path might be, or whether I set the token wrong.

But I can't get `make_project_table.py` to run.

**Comment by mkolopanis** [comment]
hera-validation is a public repo now right? does this still require the credentials?

**Comment by steven-murray** [comment]
I did try without credentials and it errored for me.

**Comment by jaguirre** [comment]
Eh, I'm an idiot.  My token had a trailing `\n` apparently due to the way I created the file.

It runs.  Looks awesome.

Add a note somewhere about pygithub and making the token, and I think it's good to go.

Sorry for the delay.

**Pull request merged**